### ISSUE TITLE
Re-enable random peer networking errors

### ISSUE THREAD

**Issue by dpc** [fedimint/fedimint#4281]
In https://github.com/fedimint/fedimint/pull/4278 I'm disabling random peer errors because they expose some consensus problem that is causing severe flakiness which can be reproduced clearly with: https://github.com/fedimint/fedimint/pull/4250 . 

After the problem is addressed we should re-enable them.

**Comment by elsirion** [comment]
@joschisan I wonder if there is a failure rate threshold above which AlephBFT cannot make progress anymore. The old network stack was (over-)built to support excessively unreliable networks with Tor p2p communication in mind. AlephBFT supports network interruptions natively, but maybe not if connections close every 10 packets on average (not sure how to interpret the `0.1` failure rate in #4278)?

**Comment by dpc** [comment]
> but maybe not if connections close every 10 packets on average (not sure how to interpret the 0.1 failure rate in https://github.com/fedimint/fedimint/pull/4278)

Even if I added a patch limit number of random errors to like 7 per whole test, the test hangs completely. And looks like things are going very wrong even on a first error.

**Comment by joschisan** [comment]
> @joschisan I wonder if there is a failure rate threshold above which AlephBFT cannot make progress anymore. 

In my tests it got slower the more I dropped and completely stopped making progress at 50% of messages dropped.

**Comment by elsirion** [comment]
@joschisan do we still have the capability to simulate bad network conditions? I faintly remember removing that.

**Comment by dpc** [comment]
I don't think this is worth doing. It's AlephBFT job to worry about stuff like that, since we've already pulled it in, we trust that it sound and works. No need to slow everything down second guessing it. Even if alephbft did have bugs, and we managed to trigger them during our very mild and predictable testing that includes no malicious pees, we would not have enough info to tell what went wrong or reproduce.
